### PR TITLE
Add buffer visulization

### PR DESCRIPTION
**Pull request by Geenz**
Adds buffer viz via RenderBufferVisualization.  Also moves the mip generation into flush, and mipmap generation options.  Manual is currently unused.

### CODE REVIEW

File: indra/newview/app_settings/shaders/class1/deferred/postDeferredVisualizeBuffers.glsl [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/** 
+ * @file postDeferredNoDoFF.glsl
+ *
+ * $LicenseInfo:firstyear=2007&license=viewerlgpl$
+ * Second Life Viewer Source Code
+ * Copyright (C) 2007, Linden Research, Inc.
```
Comment (RunitaiLinden) [inline indra/newview/app_settings/shaders/class1/deferred/postDeferredVisualizeBuffers.glsl:6]: Should fix firstyear and file name

File: indra/newview/app_settings/shaders/class1/deferred/postDeferredVisualizeBuffers.glsl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#extension GL_ARB_texture_rectangle : enable
+
+/*[EXTRA_CODE_HERE]*/
+
+#ifdef DEFINE_GL_FRAGCOLOR
+out vec4 frag_color;
+#else
+#define frag_color gl_FragColor
+#endif
+
```
Comment (RunitaiLinden) [inline indra/newview/app_settings/shaders/class1/deferred/postDeferredVisualizeBuffers.glsl:35]: I know it's copy/paste, but this syntax is deprecated, you can just do "out vec4 frag_color"

File: scripts/messages/message_template.msg.sha1 [hunk 3]
```diff
@@ -1 +1 @@
-dddb11f7e45f1779ff536819f36a20e63d572ba8
\ No newline at end of file
+dd15c52581b3fe99e072b26872deba2560893fc4
```
Comment (RunitaiLinden) [inline scripts/messages/message_template.msg.sha1:1]: Erm... uh oh.... revert?

**Review by RunitaiLinden — CHANGES_REQUESTED** [review]

**Comment by RunitaiLinden** [comment]
Mostly bikeshedding comments, but definitely revert that message template hash bump before merging

**Comment by Geenz** [comment]
Merging for now.  Dave brings up a good point as far as deprecated usage of frag_color.  Will update it in line where I come across it.

**Pull request merged**